include/seqan3/search/algorithm/detail/search_scheme_algorithm.hpp

```diff
@@ -0,0 +1,531 @@
+//!\endcond
+
+/*!\brief Searches a query sequence in a bidirectional index using a single search of a search schemes.
+ *        Sub-function for searching the remaining part of the current block without any errors.
+ * \tparam abort_on_hit If the flag is set, the search aborts on the first hit.
+ * \param[in] it Iterator of atring index built on the text that will be searched.
+ * \param[in] query Query sequence to be searched with the iterator.
+ * \param[in] lb Left bound of the infix of `query` already searched (exlusive).
+ * \param[in] rb Right bound of the infix of `query` already searched (exlusive).
+ * \param[in] errors_spent Number of errors spent while searching the infix of `query`.
+ * \param[in] block_id Id of the block that infix is extended in next.
+ * \param[in] go_right The infix will be extended to the right if the flag is set to true.
+ * \param[in] search Search of a search scheme to be used for searching.
+ * \param[in] blocks_length Cumulative block lengths of the search.
+ * \param[in] error_left Number of errors left for matching the remaining suffix of the query sequence.
+ * \param[in] delegate Function that is called on every hit. Takes `index::iterator_type` as argument.
+ */
+template <bool abort_on_hit>
+inline bool search_ss_exact(auto it, auto & query, uint64_t const lb, uint64_t const rb, uint8_t const errors_spent,
+                            uint8_t const block_id, bool const go_right, auto const & search, auto const & blocks_length,
+                            search_param const error_left, auto && delegate)
+{
+    uint8_t const block_id2 = std::min<uint8_t>(block_id + 1, search.blocks() - 1);
+    bool const go_right2 = (block_id < search.blocks() - 1) && search.pi[block_id + 1] > search.pi[block_id];
+
+    if (go_right)
+    {
+        uint64_t const infix_lb = rb - 1; // inclusive
+        uint64_t const infix_rb = lb + blocks_length[block_id] - 1; // exclusive
+
+        if (!it.extend_right(query | ranges::view::slice(infix_lb, infix_rb + 1)))
+            return false;
+
+        if (search_ss<abort_on_hit>(it, query, lb, infix_rb + 2, errors_spent, block_id2, go_right2, search, blocks_length,
+                                    error_left, delegate) && abort_on_hit)
+        {
+            return true;
+        }
+    }
+    else
+    {
+        uint64_t const infix_lb = rb - blocks_length[block_id] - 1; // inclusive
+        uint64_t const infix_rb = lb - 1; // inclusive
+
+        if (!it.extend_left(query | ranges::view::slice(infix_lb, infix_rb + 1)))
+            return false;
+
+        if (search_ss<abort_on_hit>(it, query, infix_lb, rb, errors_spent, block_id2, go_right2, search, blocks_length,
+                                    error_left, delegate) && abort_on_hit)
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+/*!\brief Searches a query sequence in a bidirectional index using a single search of a search schemes.
```

Comment:
```
/*!\brief Searches a query sequence in a bidirectional index using a single search of a search schemes.
              Sub-function for deletions at the end of a block.
* \copydetails search_ss_exact
```
Should work (maybe you need the `search_ss_exact` with the namespace. Otherwise apply review from `search_ss_exact`.